app/scripts/views/mixins/flow-events-mixin.js

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Populates the flow model and initialises flow event handlers.
+
+define(function (require, exports, module) {
+  'use strict';
+
+  const Flow = require('models/flow');
+
+  module.exports = {
+    afterRender () {
+      this.flow = new Flow({
+        sentryMetrics: this.sentryMetrics,
+        window: this.window
+      });
+
+      const flowId = this.flow.get('flowId');
+      const flowBegin = this.flow.get('flowBegin');
+
+      this.metrics.setFlowEventMetadata({
+        flowBeginTime: flowBegin,
+        flowId: flowId
+      });
+    },
+
+    // Emit an engage flow event when an input gains focus. However,
+    // don't emit that flow event for the initial autofocus action,
+    // and also don't emit it when the entire window is hidden/shown.
+    afterVisible () {
+      const onInputFocus = () => {
+        this._engageFlowEventsForm();
+      };
+
+      const onWindowFocus = () => {
+        setTimeout(() => {
```

Comment:
Could this section be removed by using `click` and `touchstart` handler on input elements?

```diff
diff --git a/app/scripts/views/mixins/flow-events-mixin.js b/app/scripts/views/mixins/flow-events-mixin.js
index 1a30425..e61760e 100644
--- a/app/scripts/views/mixins/flow-events-mixin.js
+++ b/app/scripts/views/mixins/flow-events-mixin.js
@@ -25,34 +25,12 @@ define(function (require, exports, module) {
       });
     },
 
-    // Emit an engage flow event when an input gains focus. However,
-    // don't emit that flow event for the initial autofocus action,
-    // and also don't emit it when the entire window is hidden/shown.
-    afterVisible () {
-      const onInputFocus = () => {
-        this._engageFlowEventsForm();
-      };
-
-      const onWindowFocus = () => {
-        setTimeout(() => {
-          this.$el.on('focus', 'input', onInputFocus);
-        }, 0);
-        this.window.removeEventListener('focus', onWindowFocus);
-      };
-
-      const onWindowBlur = () => {
-        this.$el.off('focus', 'input', onInputFocus);
-        this.window.addEventListener('focus', onWindowFocus);
-      };
-
-      this.$el.on('focus', 'input', onInputFocus);
-      this.window.addEventListener('blur', onWindowBlur);
-    },
-
     events: {
       'click a': '_clickFlowEventsLink',
+      'click input': '_engageFlowEventsForm',
       'input input': '_engageFlowEventsForm',
-      'submit': '_submitFlowEventsForm'
+      'submit': '_submitFlowEventsForm',
+      'touchstart input': '_engageFlowEventsForm'
     },
 
     _clickFlowEventsLink (event) {
```